lib/internal/Magento/Framework/Dto/DtoProcessor/GetHydrationStrategy.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Framework\Dto\DtoProcessor;
+
+use LogicException;
+use Magento\Framework\Api\SimpleDataObjectConverter;
+use Magento\Framework\Reflection\NameFinder;
+use ReflectionException;
+use Zend\Code\Reflection\ClassReflection;
+
+class GetHydrationStrategy
+{
+    /**
+     * Strategy for setter hydration
+     */
+    public const HYDRATOR_STRATEGY_SETTER = 'setter';
+
+    /**
+     * Strategy for constructor parameters injection
+     */
+    public const HYDRATOR_STRATEGY_CONSTRUCTOR_PARAM = 'constructor';
+
+    /**
+     * Strategy for constructor data parameter injection
+     */
+    public const HYDRATOR_STRATEGY_CONSTRUCTOR_DATA = 'data';
+
+    /**
+     * List of orphan parameters
+     */
+    public const HYDRATOR_STRATEGY_ORPHAN = 'orphan';
+
+    /**
+     * @var NameFinder
+     */
+    private $nameFinder;
+
+    /**
+     * @var DtoReflection
+     */
+    private $dtoReflection;
+
+    /**
+     * @param NameFinder $nameFinder
+     * @param DtoReflection $dtoReflection
+     */
+    public function __construct(
+        NameFinder $nameFinder,
+        DtoReflection $dtoReflection
+    ) {
+        $this->nameFinder = $nameFinder;
+        $this->dtoReflection = $dtoReflection;
+    }
+
+    /**
+     * Return the strategy for values injection.
+     *
+     *
+     * @param string $className
+     * @param array $data
+     * @return array
+     * @throws ReflectionException
+     */
+    public function execute(string $className, array $data): array
```

Comment:
I would like to propose splitting up this method into multiple for better readability.